src/TmdbServiceProvider.php

```diff
@@ -74,12 +83,44 @@ public function register()
             $config = $this->provider->config();
             $options = $config['options'];
 
+            $ed = $this->app->make('Tmdb\Laravel\Adapters\EventDispatcherAdapter');
+
             // Use an Event Dispatcher that uses the Laravel event dispatcher
-            $options['event_dispatcher'] = $this->app->make('Tmdb\Laravel\Adapters\EventDispatcherAdapter');
+            $options['event_dispatcher']['adapter'] = $ed;
+
+            if (($config['bearer_token'] ?? '') !== '') {
+                $options['api_token'] = new BearerToken($config['bearer_key']);
+            } else if (($config['api_key'] ?? '') !== '') {
+                $options['api_token'] = new ApiToken($config['api_key']);
+            } else {
+                throw new \RuntimeException("php-tmdb-laravel requires bearer_token or api_key");
+            }
+
+
+            unset($options['log']);
+            unset($options['cache']);
+
+            $client = new Client($options);
+
+            /**
+             * Required event listeners and events to be registered with the PSR-14 Event Dispatcher.
+             */
+            $requestListener = new RequestListener($client->getHttpClient(), $ed);
+            $ed->addListener(RequestEvent::class, $requestListener);
+
+            $apiTokenListener = new ApiTokenRequestListener($client->getToken());
+            $ed->addListener(BeforeRequestEvent::class, $apiTokenListener);
+
+            $acceptJsonListener = new AcceptJsonRequestListener();
+            $ed->addListener(BeforeRequestEvent::class, $acceptJsonListener);
+
+            $jsonContentTypeListener = new ContentTypeJsonRequestListener();
+            $ed->addListener(BeforeRequestEvent::class, $jsonContentTypeListener);
+
+            $userAgentListener = new UserAgentRequestListener();
+            $ed->addListener(BeforeRequestEvent::class, $userAgentListener);
 
```

Comment:
You might wanna also add the other listeners here via configuration, it's optional though.